haystack/pipelines/base.py

```diff
@@ -619,6 +697,110 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
         with open(path, 'w') as outfile:
             yaml.dump(config, outfile, default_flow_style=False)
 
+    
+    def _format_document_answer(self, document_or_answer: dict):
+        return "\n \t".join([f'{name}: {value}' for name, value in document_or_answer.items()])
+
+    def _format_wrong(self, query: dict):
+        metrics = "\n \t".join([f'{name}: {value}' for name, value in query['metrics'].items()])
+        documents = "\n\n \t".join([self._format_document_answer(doc) for doc in query.get('documents', [])])
+        documents = f"Documents: \n \t{documents}\n" if len(documents) > 0 else ""
+        answers = "\n\n \t".join([self._format_document_answer(answer) for answer in query.get('answers', [])])
+        answers = f"Answers: \n \t{answers}\n" if len(answers) > 0 else ""
+        gold_document_ids = "\n \t".join(query['gold_document_ids'])
+        gold_answers = "\n \t".join(query.get('gold_answers', []))
+        gold_answers = f"Gold Answers: \n \t{gold_answers}\n" if len(gold_answers) > 0 else ""
+        s = (
+            f"Query: \n \t{query['query']}\n"
+            f"{gold_answers}"
+            f"Gold Document Ids: \n \t{gold_document_ids}\n"
+            f"Metrics: \n \t{metrics}\n"
+            f"{answers}"
+            f"{documents}"
+            f"_______________________________________________________"
+        )
+        return s
+
+    def _format_wrongs_node(self, node_name: str, wrongs_formatted: str):
+        s = (
+            f"                Wrong {node_name} Examples\n"
+            f"=======================================================\n"
+            f"{wrongs_formatted}\n"
+            f"=======================================================\n"
+        )
+        return s
+
+    def _format_wrongs_report(self, eval_result: EvaluationResult, n_wrong_examples: int = 3):
```

Comment:
Similar question here: This code seems to generic enough to reuse it for correct examples so maybe we should rename it? Something like `_format_examples_report`?

Reply:
Maybe we leave that change for a separate issue and pull request.